RHIDP-2908 + RHIDP-2895: Remove k8s from OCP section, split k8s by platform (#328)
redhat-developer/red-hat-developers-documentation-rhdh: modules/admin/proc-rhdh-deploy-aks-helm.adoc

az login [--tenant=<optional_directory_name>]
----

. Create a resource group by running the following command:
+
[source,terminal]
----
az group create --name <resource_group_name> --location <location>
----
+
[TIP]
====
You can list available regions by running the following command:

[source,terminal]
----
az account list-locations -o table
----

====

. Create an {aks-short} cluster by running the following command:
+
[source,terminal]
----
az aks create \
--resource-group <resource_group_name> \
--name <cluster_name> \
--enable-managed-identity \
--generate-ssh-keys
----
+
You can refer to `--help` for additional options.

. Connect to your cluster by running the following command:
+
[source]
----
az aks get-credentials --resource-group <resource_group_name> --name <cluster_name>
----
+
The previous command configures the Kubernetes client and sets the current context in the `kubeconfig` to point to your AKS cluster.

. Open terminal and run the following command to add the Helm chart repository:
+
[source,terminal]
----
helm repo add openshift-helm-charts https://charts.openshift.io/
----

. Create and activate the _<rhdh>_ namespace:
+
[source,terminal,subs="quotes+"]
----

> NAMESPACE=_<rhdh>_

Reviewer: ```suggestion
DEPLOYMENT_NAME=_redhat-developer-hub_
NAMESPACE=_<rhdh>_
```

kubectl create namespace ${NAMESPACE}
kubectl config set-context --current --namespace=${NAMESPACE}
----

> . To create `ImagePull Secret`, run the following command:

Reviewer: ```suggestion
. Create a pull secret, which is used to pull the {product-short} images from the {company-name} Ecosystem, by running the following command:
```

+
[source,terminal]
----
kubectl -n <namespace> create secret docker-registry rhdh-pull-secret \
    --docker-server=registry.redhat.io \
    --docker-username=<redhat_user_name> \
    --docker-password=<redhat_password> \
    --docker-email=<email>
----

. Create a file named `values.yaml` using the following template:
+
[source,yaml]
----
global:
  host: <app_address>
route:
  enabled: false
upstream:
  ingress:
    enabled: true
    className: webapprouting.kubernetes.azure.com
    host:
  backstage:
    image:
      pullSecrets:
        - rhdh-pull-secret
    podSecurityContext:
      fsGroup: 3000
  postgresql:
    image:
      pullSecrets:
        - rhdh-pull-secret
    primary:
      podSecurityContext:
        enabled: true
        fsGroup: 3000
  volumePermissions:
    enabled: true
----

> . To install Helm Chart, run the following command:

Reviewer: ```suggestion
. To install {prod-short} using the Helm Chart, run the following command:
```

+

> [source,terminal]

Reviewer: ```suggestion
[source,terminal,subs="attributes+"]
```

----

> helm -n <namespace> install -f values.yaml <deployment_name> openshift-helm-charts/redhat-developer-hub --version {product-chart-version}

Reviewer: ```suggestion
helm -n <namespace> install -f values.yaml $DEPLOYMENT_NAME openshift-helm-charts/redhat-developer-hub --version {product-chart-version}
```

----

. Verify the deployment status:
+
[source,terminal]
----

> kubectl get deploy <deployment_name>-developer-hub -n <namespace>

Reviewer: ```suggestion
kubectl get deploy $DEPLOYMENT_NAME-developer-hub -n $NAMESPACE
```

----

. Configure your {product-short} Helm chart instance with the {product-short} database password and router base URL values from your {ocp-short} cluster:
+
[source,terminal,subs="attributes+"]
----
PASSWORD=$(kubectl get secret redhat-developer-hub-postgresql -o jsonpath="{.data.password}" | base64 -d)
CLUSTER_ROUTER_BASE=$(kubectl get route console -n openshift-console -o=jsonpath='{.spec.host}' | sed 's/^[^.]*\.//')

> helm upgrade redhat-developer-hub -i "https://github.com/openshift-helm-charts/charts/releases/download/redhat-redhat-developer-hub-{product-chart-version}/redhat-developer-hub-{product-chart-version}.tgz" \

Reviewer: ```suggestion
helm upgrade <deployment_name> -i "https://github.com/openshift-helm-charts/charts/releases/download/redhat-redhat-developer-hub-{product-chart-version}/redhat-developer-hub-{product-chart-version}.tgz" \
```

    --set global.clusterRouterBase="$CLUSTER_ROUTER_BASE" \
    --set global.postgresql.auth.password="$PASSWORD"
----

. Display the running {product-short} instance URL, by running the following command:
+
[source,terminal]
----

> echo "https://redhat-developer-hub-$NAMESPACE.$CLUSTER_ROUTER_BASE"

Reviewer: ```suggestion
echo "https://$DEPLOYMENT-$NAMESPACE.$CLUSTER_ROUTER_BASE"
```

----

.Verification

* Open the running {product-short} instance URL in your browser to use {product-short}.

.Upgrade

* To upgrade the deployment, run the following command:
+

> [source,terminal]

Reviewer: ```suggestion
[source,terminal,subs="attributes+"]
```

----
helm upgrade <deployment_name> -i https://github.com/openshift-helm-charts/charts/releases/download/redhat-redhat-developer-hub-{product-chart-version}/redhat-developer-hub-{product-chart-version}.tgz
----

.Delete

* To delete the deployment, run the following command:
+
[source,terminal]
----

> helm -n <namespace> delete <deployment_name>

Reviewer: ```suggestion
helm -n $NAMESPACE delete $DEPLOYMENT_NAME
```

az login [--tenant=<optional_directory_name>]
----

. Create a resource group by running the following command:
+
[source,terminal]
----
az group create --name <resource_group_name> --location <location>
----
+
[TIP]
====
You can list available regions by running the following command:

[source,terminal]
----
az account list-locations -o table
----

====

. Create an {aks-short} cluster by running the following command:
+
[source,terminal]
----
az aks create \
--resource-group <resource_group_name> \
--name <cluster_name> \
--enable-managed-identity \
--generate-ssh-keys
----
+
You can refer to `--help` for additional options.

. Connect to your cluster by running the following command:
+
[source]
----
az aks get-credentials --resource-group <resource_group_name> --name <cluster_name>
----
+
The previous command configures the Kubernetes client and sets the current context in the `kubeconfig` to point to your AKS cluster.

. Open terminal and run the following command to add the Helm chart repository:
+
[source,terminal]
----
helm repo add openshift-helm-charts https://charts.openshift.io/
----

. Create and activate the _<rhdh>_ namespace:
+
[source,terminal,subs="quotes+"]
----

> DEPLOYMENT_NAME=_redhat-developer-hub_

Reviewer: ```suggestion
DEPLOYMENT=_<redhat-developer-hub>_
NAMESPACE=_<rhdh>_
```

+
[source,terminal]
----
kubectl -n $NAMESPACE create secret docker-registry rhdh-pull-secret \
    --docker-server=registry.redhat.io \
    --docker-username=<redhat_user_name> \
    --docker-password=<redhat_password> \
    --docker-email=<email>
----

. Create a file named `values.yaml` using the following template:
+
[source,yaml]
----
global:
  host: <app_address>
route:
  enabled: false
upstream:
  ingress:
    enabled: true
    className: webapprouting.kubernetes.azure.com
    host:
  backstage:
    image:
      pullSecrets:
        - rhdh-pull-secret
    podSecurityContext:
      fsGroup: 3000
  postgresql:
    image:
      pullSecrets:
        - rhdh-pull-secret
    primary:
      podSecurityContext:
        enabled: true
        fsGroup: 3000
  volumePermissions:
    enabled: true
----

. To install {product-short} by using the Helm chart, run the following command:
+
[source,terminal,subs="attributes+"]
----
helm -n $NAMESPACE install -f values.yaml $DEPLOYMENT_NAME openshift-helm-charts/redhat-developer-hub --version {product-chart-version}
----

. Verify the deployment status:
+
[source,terminal]
----

> kubectl get deploy $DEPLOYMENT_NAME-developer-hub -n $NAMESPACE

Reviewer: ```suggestion
kubectl get deploy $DEPLOYMENT_NAME -n $NAMESPACE
```

----

> . Configure your {product-short} Helm chart instance with the {product-short} database password and router base URL values from your {ocp-short} cluster:

Reviewer: ```suggestion
. Configure your {product-short} Helm chart instance with the {product-short} database password and router base URL values from your cluster:
```

+
[source,terminal,subs="attributes+"]
----
PASSWORD=$(kubectl get secret redhat-developer-hub-postgresql -o jsonpath="{.data.password}" | base64 -d)
CLUSTER_ROUTER_BASE=$(kubectl get route console -n openshift-console -o=jsonpath='{.spec.host}' | sed 's/^[^.]*\.//')

> helm upgrade $DEPLOYMENT -i "https://github.com/openshift-helm-charts/charts/releases/download/redhat-redhat-developer-hub-{product-chart-version}/redhat-developer-hub-{product-chart-version}.tgz" \

Reviewer: ```suggestion
helm upgrade $DEPLOYMENT_NAME -i "https://github.com/openshift-helm-charts/charts/releases/download/redhat-redhat-developer-hub-{product-chart-version}/redhat-developer-hub-{product-chart-version}.tgz" \
```

    --set global.clusterRouterBase="$CLUSTER_ROUTER_BASE" \
    --set global.postgresql.auth.password="$PASSWORD"
----

. Display the running {product-short} instance URL, by running the following command:
+
[source,terminal]
----

> echo "https://$DEPLOYMENT-$NAMESPACE.$CLUSTER_ROUTER_BASE"

Reviewer: ```suggestion
echo "https://$DEPLOYMENT_NAME-$NAMESPACE.$CLUSTER_ROUTER_BASE"
```

> ----

Author: @nickboldt should these be `$DEPLOYMENT_NAME` instead of `$DEPLOYMENT`?
